store/store.go

```diff
@@ -908,7 +940,13 @@ func (s *Store) fakeChannels(snapID string, user *auth.UserState) (map[string]*s
 		Data:        jsonData,
 	}
 
-	resp, err := s.retryRequest(context.TODO(), s.client, reqOptions, user)
+	var results struct {
+		Payload struct {
+			SnapDetails []*snapDetails `json:"clickindex:package"`
+		} `json:"_embedded"`
+	}
+
+	resp, err := s.retryRequestDecode(context.TODO(), s.client, reqOptions, user, decodeStrategy{decodeOnHTTP200, &results, nil, nil})
```

Comment:
200 and 201 can both be taken as success inside the function itself. The last one can also decode the body as the error case as long as it's not empty, I believe. Then, if we do have cases that are indeed more complex (assertions), we can use the organization at https://gist.github.com/niemeyer/1bbfa953f4dc4cb91db78e922332f116